src/ImageSharp/Formats/Jpg/JpegEncoderCore.cs

```diff
@@ -484,29 +429,170 @@ private void EmitHuffRLE(HuffIndex index, int runLength, int value)
         }
 
         /// <summary>
+        /// Encodes the image with no subsampling.
+        /// </summary>
+        /// <typeparam name="TColor">
+        /// The pixel format.
+        /// </typeparam>
+        /// <param name="pixels">
+        /// The pixel accessor providing access to the image pixels.
+        /// </param>
+        private void Encode444<TColor>(PixelAccessor<TColor> pixels)
+            where TColor : struct, IPackedPixel, IEquatable<TColor>
+        {
+            // TODO: Need a JpegEncoderScanProcessor<TColor> struct to encapsulate all this mess:
```

Comment:
This work is in progress for the decoder, I will describe all the details in my forcoming PR in comments & .MD. 
The changes I want to implement in the encoder will follow the same logic.